src/SFML/Window/Win32/JoystickImpl.cpp

```diff
@@ -152,6 +202,12 @@ namespace priv
 ////////////////////////////////////////////////////////////
 void JoystickImpl::initialize()
 {
+    // Try to initialize DirectInput
+    initializeDInput();
+
+    if (!directInput)
+        err() << "DirectInput not available, falling back to Windows joystick API" << std::endl;
```

Comment:
If someone doesn't want to use a joystick and doesn't have direct input available will this still output the error?

Reply:
People have already "complained" a few times about not being able to disable joystick polling, so yes... this error will still be output as soon as window events are pumped. Realistically, these can only be Windows 98 or older systems, so if they get the rest of SFML running on those kinds of systems they deserve a medal.